.github/workflows/nvm.yml

```diff
@@ -5,9 +5,47 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - run: node --version
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
       - run: |
           nvm install --latest-npm --no-progress
-          echo "::set-env name=PATH::${NVM_BIN}:${PATH}"
+          echo "$(dirname $(which node))" >> $GITHUB_PATH
         shell: bash --login {0}
       - run: node --version
+
+      # Optional: Cache npm dependencies
+      - name: Get npm cache directory
+        id: npm-cache-dir
+        shell: bash
+        run: echo "dir=$(npm config get cache)" >> $GITHUB_OUTPUT
+      - uses: actions/cache@v4
+        id: npm-cache
+        with:
+          path: ${{ steps.npm-cache-dir.outputs.dir }}
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      # Optional: Cache Yarn v1 dependencies
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "dir=$(yarn cache dir)" >> $GITHUB_OUTPUT
+      - uses: actions/cache@v4
+        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      # Optional: Cache pnpm store
+      - name: Get pnpm store directory
+        shell: bash
+        run: |
+          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV
```

Comment:
[ask] Do we need to install `pnpm`? It doesn't seem to be included in the pre-installed software list:
https://github.com/actions/runner-images/blob/ubuntu22/20240804.1/images/ubuntu/Ubuntu2204-Readme.md

Reply:
Yes, if we want this action to be runnable out of the box

We can use pnpm/action-setup https://github.com/pnpm/action-setup

Reply:
I didn't know that action. Let's include it!

Reply:
done in 219c6ee92f90a8618ecec7270167b7b2dd0d7da1